### ISSUE TITLE
Suggestion: Enable local watchface JSON config on Pebble

### ISSUE THREAD

**Issue by Miq1** [Freeyourgadget/Gadgetbridge#977]
As there are more and more watchfaces whose configuration web pages are not available any more:
would it be possible 

- to automatically extract the JSON configuration names from the PBW
- enable input of values for each of these JSON names
- send the configuration to the watchface

That could save the life for numerous watchfaces.

**Comment by danielegobbetti** [comment]
Technically it is possible, but there are two problems:
- the keys we get from the pbw are (always, iirc) numeric. The meaning of each key has to be guessed
- we don't get the type (string, int, bool) of each key

In any case I believe the functionality you mention makes sense (in case a watchapp configuration page disappears and you saved the settings within GB you can replay those at least :-) )

**Comment by danielegobbetti** [comment]
One further problem that we already faced: each time an application is updated, every key identifier might change

**Comment by pablob127** [comment]
Regarding the last problem Daniele mentioned, if a configuration webpage goes away permanently there is a big chance that the application will not be updated... :-)

**Comment by Miq1** [comment]
If it was possible at least to provide IDs and values on a debug page, one could hack or reverse engineer the possible configuration values. Then a per-app matching list with IDs, parameter names, explanations and values could be maintained that than will help replacing the external configuration sites by an internal Gadgetbridge function.

**Comment by danielegobbetti** [comment]
The returned values of a (working) configuration page are shown by our internal webview and could be saved (beware: there is only one "slot" per watchapp, last saved configuration overwrites older data).

**Comment by Miq1** [comment]
Of course, but there are several non-existent pages already where that approach will not work...

**Comment by white-gecko** [comment]
@Miq1 maybe you can go through all apps and archive the configuration pages so we will still have them for the future, for those watchfaces, where the configuration is not gone yet.
